### DISCUSSION TITLE
LLaMA

### DISCUSSION THREAD

**Discussion by nikita322** [tteck/Proxmox#3215] [Ideas]
Hi, please add script to install LLaMA (or oLLaMA)

**Comment by ulmentflam** [comment]
I don't believe @tteck wants to support scripts that require Nvidia GPU passthrough. I have a branch with an ollama script you are welcome to use, just use it at your own risk. It does take some time to setup Nvidia passthrough on LXC, but there are multiple guides available. https://github.com/ulmentflam/Proxmox/tree/experimental

**Reply by tteck** [reply]
The repository has several iGPU passthrough scripts, none NVIDIA.

**Reply by tteck** [reply]
Eight to be exact.
https://github.com/tteck/Proxmox/blob/53b24ef7b77f38a77d8ba3ba4b7bd6c458d4f1d5/misc/build.func#L594

**Reply by ulmentflam** [reply]
So would you be willing to support dedicated GPU passthrough for Nvidia if there was an optional driver install script for the PVE Shell, then an update to the build step to pass the GPU through if the driver is available with a similar methodology to the above? For some context; in my Proxmox setup I don't have an integrated graphics card to pass through, so I am doing all my hardware acceleration through dedicated Nvidia cards that are passed through for hardware acceleration, transcoding, etc...

**Reply by tteck** [reply]
No, you have to modify the Proxmox host.

**Reply by ulmentflam** [reply]
Got it. Now I think about it, multiple steps in the process require you edit host files. VFIO was simpler but you can't mount that to multiple LXC for good reason. 🤔 Worth the discussion though, as I am sure you will have more interest in self-hosting LLMs and Diffusion models that require dedicated GPUs.

**Comment by havardthom** [comment]
Added option for installing Ollama in Open WebUI LXC script https://github.com/tteck/Proxmox/blob/main/CHANGELOG.md#2024-10-27

Alternatively there is also a separate Ollama LXC script which builds Ollama from source and supports Intel iGPU passthrough (very long install time): `bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/ollama.sh)"`
